CodeInsights: Use wildcard `Popover` in the code insights components

vovakulikov opened the pull request:
## PRs map 

Based on [Adding `Popover` component](https://github.com/sourcegraph/sourcegraph/pull/29791)
**PLEASE BE SURE THAT BASED PR IS MERGED BEFORE MERGING THIS ONE**

This PR is the second PR of the series of changes for the `Popover` component. Previous these changes were in [one big PR here ](https://github.com/sourcegraph/sourcegraph/pull/29518)

1. [Add tether sub package PR](https://github.com/sourcegraph/sourcegraph/pull/29790)
2. [Adding `Popover` component](https://github.com/sourcegraph/sourcegraph/pull/29791)
3. _PRs from the third phase could be merged in parallel and reviewed independently_ 
   3.1 [Use `Popover` component in `Menu button` instead of built-in reach UI popover](https://github.com/sourcegraph/sourcegraph/pull/29792)
   3.2 **This PR** Migrate Code Insights to new wildcard `Popover` component
   
## Context 

This PR removes the old code insights specific component and uses a brand new wildcard `Popover` component instead.

vovakulikov commented:
@AlicjaSuska I've noticed that our contrast between cards and popover element background isn't enough. It might be possible to confuse users because popover now has the same background as code insights cards have. 


My suggestion is let's change the background of the drill-down popover element, compare these two 

<img width="1255" alt="Screenshot 2022-01-16 at 22 23 43" src="https://user-images.githubusercontent.com/18492575/149674873-5784ef69-9a78-4040-b3f5-37f2450567f0.png">

<img width="1255" alt="Screenshot 2022-01-16 at 22 26 45" src="https://user-images.githubusercontent.com/18492575/149674878-b2daf60a-bbe0-4074-b45d-e87599727fe5.png">

Also notice that I removed the Regexp header in the drill-down popover because it seems unnecessary at least for now when we have only one option for drill down - regexp.

AlicjaSuska commented:
Thank you @vovakulikov:
- background looks good, I'm ok with this change
- I would recommend leaving the 'Regular expression' header. Now, there is no information in the interface that regex should be used. This was also the reasoning behind the initial decision of leaving the header in the popover (with no radio button ofc)

AlicjaSuska reviewed (changes requested):
Thank you @vovakulikov:

background looks good, I'm ok with this change
I would recommend leaving the 'Regular expression' header. Now, there is no information in the interface that regex should be used. This was also the reasoning behind the initial decision of leaving the header in the popover (with no radio button ofc)

vovakulikov commented:
@AlicjaSuska What do you think instead of using header regexp we put this information in the field description section (text below the input). My main concern here is that we have two headers (even three actually in one visual column) that break a little bit of visual tempo there. And they are all the same size and color. 

<img width="172" alt="Screenshot 2022-01-17 at 15 29 45" src="https://user-images.githubusercontent.com/18492575/149769875-98467a72-72a0-454d-8633-60ebcf36a2e7.png">

Solution with input description text 

<img width="923" alt="Screenshot 2022-01-17 at 15 42 01" src="https://user-images.githubusercontent.com/18492575/149771500-ac823e90-9921-4cf0-8b43-a7a375e2bb55.png">

Note that in the screenshot above we used text-muted for description text (default color text for this kind of description) and it works but if we use different tooltip background color from my previous message in this thread then I think we should use common text color for these descriptions because the contrast is pretty low with text muted texts. See the screenshot above 

<img width="919" alt="Screenshot 2022-01-17 at 15 43 05" src="https://user-images.githubusercontent.com/18492575/149771786-c63f0ba6-35f7-44a3-aa62-407c188cceab.png">

vovakulikov commented:
Maybe the problem with contrast could be solved later because we have the same thing in the creation UI at the moment. Gray muted text on a non-white background 

<img width="595" alt="Screenshot 2022-01-17 at 15 46 50" src="https://user-images.githubusercontent.com/18492575/149772007-2547f387-c483-447a-aa80-8cb4e291d919.png">

valerybugakov commented on client/web/src/enterprise/insights/components/insights-view-grid/components/backend-insight/components/drill-down-filters-action/DrillDownFiltersPanel.tsx:
```diff
@@ -45,26 +43,23 @@ export const DrillDownFiltersAction: React.FunctionComponent<DrillDownFiltersPro
-                className={classNames('btn-icon p-1', styles.filterButton, {
+                type="button"
+                aria-label={isFiltered ? 'Active filters' : 'Filters'}
+                className={classNames('btn btn-icon p-1', styles.filterButton, {
```
The `btn` class is redundant with `as={Button}`

```suggestion
                className={classNames('btn-icon p-1', styles.filterButton, {
```

valerybugakov reviewed (approved):
✅

vovakulikov commented:
Based on sync with @alicja we're ok with removing the Regular Expression header from the drill-down panel layout but we have to come up with some description text there instead (and instead of a solution with input description texts that I suggested above as well)

AlicjaSuska commented:
@vovakulikov Here is the visual and copy update ([Figma](https://www.figma.com/file/1zvF8Owrh1ndVH29lvyrXn/RFC-419%3A-Code-Insights-%E2%80%9CDrilldown%E2%80%9D?node-id=1717%3A12394)):

Please, add text: 'Use regular expression to include and exclude repositories from the scope of this insight.' above the input fields. The text should be small and text-muted color. Thank you!
<img width="918" alt="Screenshot 2022-01-20 at 16 16 22" src="https://user-images.githubusercontent.com/20326070/150367180-5dd4a00d-a609-4673-b0b7-69fae764e365.png">

vovakulikov commented:
@AlicjaSuska I added text/description above the fields but about `text-muted` color. Since early we updated the background of this popover what do you think to stick with common text color here for the description text. 

Example 
<img width="1225" alt="Screenshot 2022-01-20 at 18 44 41" src="https://user-images.githubusercontent.com/18492575/150372167-5ee11889-30c9-4222-81c1-c1fde7a65563.png">

AlicjaSuska reviewed (approved):
Thank you for making changes @vovakulikov Small improvement: instead of 16px top-bottom margin for the text, can we have 12px (it's like this in the design). Now, it feels like this part of the popover is too empty and takes too much space. Thank you!

vovakulikov commented:
@AlicjaSuska Sorry I missed this from your layout.

 I'm actually thinking that we probably don't need to have this separation line between the header and the description. Because it was needed to separate header and different filters mode before (regexp and repo list)
 
 But due to the fact that we don't have any other mode besides regexp and we introduced description block I think naturally it should be bound to the header H4 element. 
 
Also, note that I did change the 'default filters applied' copy. It should be consistent with the code intel tooltip I think since they provide status in a similar way. 
 
<img width="869" alt="Screenshot 2022-01-20 at 23 47 11" src="https://user-images.githubusercontent.com/18492575/150420136-2265055b-0cf8-40ee-9109-49897b16bc81.png">

<img width="676" alt="Screenshot 2022-01-20 at 23 38 19" src="https://user-images.githubusercontent.com/18492575/150420865-b91ba86c-5e76-460f-a170-ae4ac4b6a97c.png">

vovakulikov commented:
@AlicjaSuska This is the version with upper text in the badge. What do you think? It seems like the link at the end "Learn more" breaks the vertical rhythm. I would place it before the badge but it's also I bit strange because - Learn more about what?  Or not? 

Also, what do you think about the copy here? Should it be "Filters applied" instead of the "Default filters applied"? 

<img width="858" alt="Screenshot 2022-01-20 at 23 59 45" src="https://user-images.githubusercontent.com/18492575/150421618-3f78a35e-aed1-4f8d-a9a9-6912d357c7b1.png">




<img width="676" alt="Screenshot 2022-01-20 at 23 38 19" src="https://user-images.githubusercontent.com/18492575/150421386-e6ee118f-aca6-4471-8e76-8d0ea386fd18.png">

vovakulikov commented:
@AlicjaSuska I moved the link right next to the header of this panel.

<img width="872" alt="Screenshot 2022-01-21 at 07 09 25" src="https://user-images.githubusercontent.com/18492575/150464251-22483d33-92da-49f3-a6b3-c9329c1f7785.png">

AlicjaSuska commented:
@vovakulikov pasting my message from Slack for transparency:

The badge doesn't really serve the purpose well in this case. 'default filters applied' is information about the state but it shouldn't get that much attention as it is for code intel. In addition, when someone applies additional filters, this badge would stay the same and, as it is very strong visually, it may be interpreted as that ONLY default filters are applied. It may be very misleading.

The way information reads in this popover is as follows:
- General info: Filter repositories -> default filters applied. Learn more
- Filtering area: regex info + inputs
- Actions: buttons
by removing the divider and adding the badge the whole flow is distorted. Badge becomes disproportionally heavy to its actual importance.

'Learn more' needs to be in the context of the 'Default filters applied'. It matters where this link is placed.

Overall: I would leave the layout as it used to be (with divider, no label, etc. the state from yesterday). The one thing I've noticed we can still improve is the regex info helper text. It would be nice to have it in one line (my designs were wider than the implementation): 'Use regular expression to change the scope of this insight.

(also, please add the period at the end of the sentence. thanks!)
